ipam/ring/ring.go

```diff
@@ -348,6 +348,25 @@ func (r *Ring) OwnedRanges() (result []address.Range) {
 	return r.splitRangesOverZero(result)
 }
 
+// OwnedAndMergedRanges does the same as OwnedRanges, except that it returns
+// merged ranges.
+func (r *Ring) OwnedAndMergedRanges() (result []address.Range) {
```

Comment:
@bboreham explained why. It's subtle. And really deserves a comment.

I am in two minds as to whether refactoring this code is worthwile - either introducing a helper that takes a boolean, or making this function call `OwnedRanges` and do the merging on the result.